test/common/testlib.py

```diff
@@ -793,6 +784,15 @@ def sitter():
             self.browser = self.new_browser()
         self.tmpdir = tempfile.mkdtemp()
 
+        def sitter():
+            if opts.sit and not self.checkSuccess():
+                if self._outcome:
+                    [traceback.print_exception(*e[1]) for e in self._outcome.errors if e[1]]
+                else:
+                    self.currentResult.printErrors()
+                sit(self.machines)
+        self.addCleanup(sitter)
```

Comment:
Thanks for that fix! This was annoying indeed. Could you please split this out as separate commit?